File: leetcode/java/1281. Subtract the Product and Sum of Digits of an Integer.java
class Solution {
    public int subtractProductAndSum(int n) {
        int currentN = n;
        int productNum = 1;
        int sumNum = 0;
        
        while(currentN >= 10) {
            productNum *= (currentN % 10);
            sumNum += (currentN % 10);
            currentN = currentN / 10;
        }
        
        productNum *= currentN;
        sumNum += currentN;
        
        return productNum - sumNum;
    }
}

File: leetcode/kotlin/1281. Subtract the Product and Sum of Digits of an Integer.kt
class Solution {
    fun subtractProductAndSum(n: Int): Int {
        var currentN = n
        var productNum = 1
        var sumNum = 0
        
        while (currentN >= 10) {
            productNum *= (currentN % 10)
            sumNum += (currentN % 10)
            currentN /= 10
        }
        
        productNum *= currentN
        sumNum += currentN
        
        return productNum - sumNum
    }
}
